src/ProductTour/ProductTour.test.jsx

```diff
@@ -132,17 +127,58 @@ describe('<ProductTour />', () => {
         expect(screen.getByRole('dialog', { name: 'Checkpoint 1' })).toBeInTheDocument();
 
         // Click the dismiss button
-        const dismissButton = screen.getByRole('button', { name: 'Dismiss' });
-        expect(dismissButton).toBeInTheDocument();
+        const closeButton = screen.getByRole('button', { name: 'Close tour' });
+        expect(closeButton).toBeInTheDocument();
 
         await act(async () => {
-          await userEvent.click(dismissButton);
+          await userEvent.click(closeButton);
         });
 
         // Verify no Checkpoints have rendered
         expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
       });
 
+      it('onClick of close icon button disables tour', async () => {
+        const { rerender } = render(<ProductTourWrapper tours={[tourData]} />);
+        // Advance the tour, close icon only appears after first step
+        await act(async () => {
```

Comment:
Given that `userEvent` [already has a wrapper](https://flyingsquirrel.medium.com/when-should-i-use-act-in-react-testing-library-d7dd22a3340e) in `act` in its internal implementation, I suggest removing this additional wrapper.